control/veda_file_discovery.py

```diff
@@ -125,162 +127,199 @@ def about_video_validate(self, meta, key):
-        if meta.get_metadata('course_video_upload_token') is None:
-            return None
+        Arguments:
+            course_id: course id identifying a course run
+            course_hex: studio_hex identifying course runs
 
-        client_title = meta.get_metadata('client_video_id')
-        course_hex = meta.get_metadata('course_video_upload_token')
-        course_url = meta.get_metadata('course_key')
-        transcript_preferences = meta.get_metadata('transcript_preferences')
-        edx_filename = key.name[::-1].split('/')[0][::-1]
+        Details:
+         - if course_hex is there, try getting course with course_hex.
+         - else try making use of course_id to get the associated course
+           and if no course is associated with the course_id, try creating
+           a new course with course_name, institution, edx_classid and
+           local_storedir.
 
-        if len(course_hex) == 0:
-            return None
+        """
+        if not course_hex:
+            try:
+                course_key = CourseKey.from_string(course_id)
+            except InvalidKeyError:
+                return
+
+            course_set = Course.objects.filter(institution=course_key.org, edx_classid=course_key.course)
+            if course_set.exists():
+                course = course_set.first()
+                course_runs = course.course_runs
+                if course_id not in course_runs:
+                    course_runs.append(course_id)
+                    course.local_storedir = ','.join(course_runs)
+                    course.save()
+            else:
+                course_name = '{org} {number}'.format(org=course_key.org, number=course_key.course)
+                course = Course.objects.create(
+                    course_name=course_name,
+                    institution=course_key.org,
+                    edx_classid=course_key.course,
+                    local_storedir=course_id,
+                )
+        else:
+            try:
+                course = Course.objects.get(studio_hex=course_hex)
+            except Course.DoesNotExist:
+                return
 
-        course_query = Course.objects.filter(studio_hex=course_hex)
-        if len(course_query) == 0:
-            V = VideoProto(
-                s3_filename=edx_filename,
-                client_title=client_title,
-                file_extension='',
-                platform_course_url=course_url
-            )
+        return course
 
-            """
-            Call VAL Api
-            """
-            val_status = 'invalid_token'
-            VAC = VALAPICall(
-                video_proto=V,
-                val_status=val_status
-            )
-            VAC.call()
+    def download_video_to_working_directory(self, key, file_name, file_extension):
+        """
+        Downloads the video to working directory from S3 and
+        returns whether its successfully downloaded or not.
+        """
+        if len(file_extension) == 3:
```

Comment:
better if `len(file_extension) == FILE_EXTENSION_LENGTH`

Reply:
BTW, why we need this if condition? Looks confusing.

Reply:
This is needed in Ingest Module, Im passing it down in the end. I don't think that we'd need it if we go for refactoring ingest and onwards modules. So, just doing this for making happy the next modules